core/templates/dev/head/services/ExplorationHtmlFormatterService.js

```
@@ -33,6 +33,8 @@ oppia.factory('ExplorationHtmlFormatterService', [
           extensionTagAssemblerService.formatCustomizationArgAttrs(
             element, interactionCustomizationArgSpecs));
         element.attr('on-submit', 'submitAnswer(answer, rulesService);');
+        element.attr('last-answer',
+          'activeCard.inputResponsePairs.slice(-1)[0].learnerInput');
```

Does this work correctly if inputResponsePairs is an empty array?

---

Yes, it gives undefined which renders all if conditions as false

---

So I just tried this in the JS console and got:
```
> x = []
[]
> x.slice(-1)
[]
> x[0]
undefined
> x[0].learnerInput
Uncaught TypeError: Cannot read property 'learnerInput' of undefined
    at <anonymous>:1:6
```
so, not sure about your assertion here, or perhaps something's hiding an error somewhere. Either way, should probably do something like 'getLastAnswer()' instead.